docs/apis/es_sender.md

```diff
@@ -0,0 +1,121 @@
+# elasticsearch_sender_api
+
+The `elasticsearch_sender_api` will provide a factory that can create sender apis that can be accessed in any operation through the `getAPI` method on the operation.
+
+
+This is a [Factory API](https://terascope.github.io/teraslice/docs/packages/job-components/api/interfaces/apifactoryregistry), which can be used to fully manage api creation and configuration.
+
+
+## Elasticsearch Sender Factory API Methods
+
+### size
+
+this will return how many seperate sender apis are in the cache
+
+### get
+parameters:
+- name: String
+
+this will fetch any sender api that is associated with the name provided
+
+### getConfig
+parameters:
+- name: String
+
+this will fetch any sender api config that is associated with the name provided
+
+### create (async)
+parameters:
+- name: String
+- configOverrides: Check options below, optional
+
+this will create an instance of a sender api, and cache it with the name given. Any config provided in the second argument will override what is specified in the apiConfig and cache it with the name provided. It will throw an error if you try creating another api with the same name parameter
+
+```typescript
+    const apiManager = this.getAPI<ElasticSenderFactoryAPI>(apiName);
+    // this will return an api cached at "normalClient" and this instance will use all configurations listed on the apiConfig
+    const client = apiManager.create('normalClient')
+
+    // this will return an api cached at "overrideClient" and this instance will have an override setting the parameter index to "other_index", this will use the rest of the configurations listed in the apiConfig
+    const overrideClient = apiManager.create('overrideClient', { index: 'other_index'})
+```
+
+### remove (async)
+parameters:
+- name: String
+
+this will remove an instance of a sender api from the cache and will follow any cleanup code specified in the api code.
+
+### entries
+
+This will allow you to iterate over the cache name and client of the cache
+
+### keys
+
+This will allow you to iterate over the cache name of the cache
+
+### values
+
+This will allow you to iterate over the clients of the cache
+
+
+## Elasticsearch Sender Instance
+This is the sender class that is returned from the create method of the APIFactory. This returns an [elastic-api](https://terascope.github.io/teraslice/docs/packages/elasticsearch-api/overview).
+
+### send
+```(records: DataEntities[]) => Promise<void>```
+parameters:
+- records: an array of data-entities
+
+
+```js
+    // this will read the first 500 bytes of the file
+    const data = [
+            DataEntity.make({ some: 'data', name: 'someName', job: 'to be awesome!' })
+    ];
+
+    await api.send(data)
```

Comment:
The indentation here is odd, also there is an obvious copy and paste comment at the top.

```suggestion
await api.send([
    DataEntity.make({ some: 'data', name: 'someName', job: 'to be awesome!' })
]);
```